lightning/src/onion_message/packet.rs

```diff
@@ -331,7 +331,7 @@ impl Readable for ControlTlvs {
 			(None, None) => None,
 		};
 
-		let valid_fwd_fmt = next_hop.is_some() && path_id.is_none();
+		let valid_fwd_fmt = next_hop.is_some() && context.is_none();
```

Comment:
@TheBlueMatt The second check is only relevant for `path_id`, which since we now don't understand would result in us failing to parse (i.e., ignoring) this message as dictated by the spec:

https://github.com/lightning/bolts/blob/c562d91ace0e95bec3c6f8758969eaf3627f23c8/04-onion-routing.md?plain=1#L1532-L1536

So we should probably drop the `context` check?

Reply:
I have kept the check intact for now in [528dc64](https://github.com/lightningdevkit/rust-lightning/commit/528dc644e5b995c10f96fe36fb803bc6433580d7).
Waiting eagerly for @TheBlueMatt's thoughts on this before I remove it!

Reply:
After our offline discussion I have removed the check in [9d7a3bf](https://github.com/lightningdevkit/rust-lightning/commit/9d7a3bf8f86fbb64354ffe5511719900a134c668)
Thanks for the suggestion and discussion!